presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java

```
@@ -36,9 +36,17 @@ private PrestoSystemRequirements() {}
 
     public static void verifyJvmRequirements()
     {
-        String specVersion = StandardSystemProperty.JAVA_SPECIFICATION_VERSION.value();
-        if ((specVersion == null) || (specVersion.compareTo("1.8") < 0)) {
-            failRequirement("Presto requires Java 1.8+ (found %s)", specVersion);
+        String javaVersion = StandardSystemProperty.JAVA_VERSION.value();
+        int sep = (javaVersion == null) ? -1 : javaVersion.lastIndexOf('_');
```

This code can be simplified and the error message improved by failing early:
```java
String javaVersion = StandardSystemProperty.JAVA_VERSION.value();
if (javaVersion == null) {
    failRequirement("Java version not defined");
}
String[] versionParts = javaVersion.split("_");
if (versionParts.length != 2) {
    failRequirement("Java version has an unknown format: %s", javaVersion);
}
String majorVersion = versionParts[0];
String minorVersion = versionParts[1];